integration-tests/mongodb-client/src/main/java/io/quarkus/it/mongodb/BookCodec.java

```diff
@@ -9,14 +9,14 @@
 import org.bson.codecs.DecoderContext;
 import org.bson.codecs.EncoderContext;
 
-import com.mongodb.MongoClient;
+import com.mongodb.MongoClientSettings;
 
 public class BookCodec implements CollectibleCodec<Book> {
 
     private final Codec<Document> documentCodec;
 
     public BookCodec() {
-        this.documentCodec = MongoClient.getDefaultCodecRegistry().get(Document.class);
+        this.documentCodec = MongoClientSettings.getDefaultCodecRegistry().get(Document.class);
```

Comment:
This might be considered as a breaking change as the application code need to be updated.

Reply:
Yes, it is described in the PR and I added the breaking change label.
The extension is still in preview so I think we can have this kind of small breaking change but if youre against it I'm OK.
In fact, it's not really a breaking change as the user can include the legacy driver that still contains this class.